.github/actions/deploy-ubuntu/action.yml

```diff
@@ -113,36 +116,49 @@ runs:
       run: |
         git --version
         $PREFIX-gcc --version
-        mvn -version
         cmake --version
+        gradle --version
+        mvn -version
         gpg --version
         df -h
 
+        export MAKEJ=$(getconf _NPROCESSORS_ONLN)
         echo Fetching $GITHUB_REPOSITORY@$GITHUB_SHA
         git init
         git fetch --depth 1 https://github.com/$GITHUB_REPOSITORY $GITHUB_SHA
         git checkout $GITHUB_SHA
+        git submodule update --init --recursive
+        git submodule foreach --recursive 'git reset --hard'
 
         echo "Fixing HOME to /root (was '$HOME')"
         export HOME=/root
-        mkdir -p $HOME/.m2
-        [[ "${{ github.event_name }}" == "pull_request" ]] && MAVEN_PHASE=install || MAVEN_PHASE=deploy
-        echo "<settings>" > $HOME/.m2/settings.xml
-        echo "<mirrors><mirror><id>google-maven-central</id><name>GCS Maven Central mirror</name><url>https://maven-central.storage-download.googleapis.com/maven2/</url><mirrorOf>central</mirrorOf></mirror></mirrors>" >> $HOME/.m2/settings.xml
-        echo "<servers><server><id>sonatype-nexus-snapshots</id><username>$CI_DEPLOY_USERNAME</username><password>$CI_DEPLOY_PASSWORD</password></server><server><id>sonatype-nexus-staging</id><username>$CI_DEPLOY_USERNAME</username><password>$CI_DEPLOY_PASSWORD</password></server></servers>" >> $HOME/.m2/settings.xml
-        echo "</settings>" >> $HOME/.m2/settings.xml
-
-        SECONDS=$(( RANDOM % 300 ))
-        echo Sleeping $SECONDS seconds to help avoid race conditions with Maven
-        sleep $SECONDS
 
-        export MAKEJ=$(getconf _NPROCESSORS_ONLN)
-        echo Executing Maven $MAVEN_PHASE on $MAKEJ processors
-        export EXT=${{ matrix.ext }}
-        export EXT2=${EXT:1}
-        export MAVEN_OPTIONS="clean $MAVEN_PHASE -B -U -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.http.retryHandler.count=3 -Djavacpp.platform=$CI_DEPLOY_PLATFORM -Djavacpp.platform.extension=$EXT $MAVEN_OPTIONS"
-        for i in {1..5}; do STATUS=0; mvn $MAVEN_OPTIONS -f $CI_DEPLOY_MODULE && break || STATUS=$? && sleep 60; done; (exit $STATUS)
-        for i in {1..5}; do STATUS=0; mvn $MAVEN_OPTIONS -f $CI_DEPLOY_MODULE/platform/$EXT2/pom.xml && break || STATUS=$? && sleep 60; done; (exit $STATUS)
+        if [[ -n ${CI_DEPLOY_NEED_GRADLE:-} ]]; then
+          [[ "${{ github.event_name }}" == "pull_request" ]] && GRADLE_TASK=publishToMavenLocal || GRADLE_TASK=publish
+          echo Executing Gradle $GRADLE_TASK ${{ matrix.options }} on $MAKEJ processors
+          export GRADLE_OPTIONS="-Dorg.gradle.jvmargs=-Xmx2048m -PjavacppPlatform=$CI_DEPLOY_PLATFORM -PjavacppPlatformExtension=${{ matrix.ext }} ${{ matrix.options }} --info"
+          gradle clean build $GRADLE_OPTIONS -b $CI_DEPLOY_MODULE/build.gradle
```

Comment:
It seems like the `-b $CI_DEPLOY_MODULE/build.gradle` bits are breaking the builds on presets where the build.gradle file is in the root directory of the repository. For example, there is no `gcc` directory inside the gcc preset which is why the build will fail for that preset.

Perhaps there should be an option/switch for this? eg `CI_DEPLOY_GRADLE_ROOT_DIRECTORY` or something similar to toggle this `-b` option?

Reply:
Just set `CI_DEPLOY_MODULE: "."`?

Reply:
That works, but wouldn't that defeat the purpose of module-specific conditions in the composite action file?

It'll work for now, but say we had a large dependency for gcc which we would install only for the gcc module, then the typical `if [[ $CI_DEPLOY_MODULE=="gcc" ]];` wouldn't work anymore.

Reply:
Hum, I suppose we should use the name of the workflow `${{ github.workflow }}` for those conditions. Does that work?

Reply:
You mean something like this? Replacing `$CI_DEPLOY_MODULE` with `${{ github.workflow}}` inside the conditionals?

https://gist.github.com/supergrecko/2632d11089b4f51f376f2c8f9a9f1b8f#file-action-ubunty-yml-L107

Reply:
Yes, like that.

Reply:
But it doesn't work if we use the workflows for something else and end up with names like `java` anyway, so I don't know. Let's leave that for another time.